src/coreclr/jit/lowerxarch.cpp

```diff
@@ -5026,20 +5028,21 @@ GenTree* Lowering::LowerHWIntrinsicDot(GenTreeHWIntrinsic* node)
         tmp2 = comp->gtClone(tmp1);
         BlockRange().InsertAfter(tmp1, tmp2);
 
-        tmp3 = comp->gtNewSimdGetUpperNode(TYP_SIMD16, tmp2, simdBaseJitType, simdSize);
+        tmp3 = comp->gtClone(tmp2);
         BlockRange().InsertAfter(tmp2, tmp3);
-        LowerNode(tmp3);
 
-        tmp1 = comp->gtNewSimdGetLowerNode(TYP_SIMD16, tmp1, simdBaseJitType, simdSize);
-        BlockRange().InsertAfter(tmp3, tmp1);
-        LowerNode(tmp1);
+        idx = comp->gtNewIconNode(0x01, TYP_INT);
+        BlockRange().InsertAfter(tmp3, idx);
 
-        tmp2 = comp->gtNewSimdBinOpNode(GT_ADD, TYP_SIMD16, tmp3, tmp1, simdBaseJitType, 16);
-        BlockRange().InsertAfter(tmp1, tmp2);
+        NamedIntrinsic permute2x128 = (simdBaseType == TYP_DOUBLE) ? NI_AVX_Permute2x128 : NI_AVX2_Permute2x128;
```

Comment:
the debug assert only checks AVX: `assert(comp->compIsaSupportedDebugOnly(InstructionSet_AVX));` in this case, should it be updated to avx2?

Reply:
We have an assert a bit higher up already. We could replicate it here if you really want it to be.